File: TheaterSuggestions/Java/SeatsSuggestions/src/main/java/com/octo/technology/SeatsSuggestions/SuggestionMade.java
package com.octo.technology.SeatsSuggestions;

import com.google.common.collect.ImmutableList;

import java.util.List;
import java.util.stream.Collectors;

public class SuggestionMade {

    private final ImmutableList<Seat> suggestedSeats;
    private final int partyRequested;
    private final PricingCategory pricingCategory;

    public SuggestionMade(List<Seat> suggestedSeats, int partyRequested, PricingCategory pricingCategory) {
        this.suggestedSeats = ImmutableList.copyOf(suggestedSeats);
        this.partyRequested = partyRequested;
        this.pricingCategory = pricingCategory;
    }

    public List<String> seatNames() {
        return suggestedSeats.stream().map(Seat::toString).collect(Collectors.toList());
    }

    public boolean MatchExpectation() {
        return suggestedSeats.size() == partyRequested;
    }

    public PricingCategory pricingCategory() {
        return pricingCategory;
    }
}

File: TheaterSuggestions/CSharp/SeatsSuggestions/SuggestionMade.cs
using System.Collections.Generic;
using System.Linq;

namespace SeatsSuggestions
{
    /// <summary>
    ///     Occurs when a Suggestion is made.
    /// </summary>
    public class SuggestionMade(int partyRequested, PricingCategory pricingCategory, List<Seat> seats)
    {
        public int PartyRequested { get; } = partyRequested;
        public PricingCategory PricingCategory { get; } = pricingCategory;

        public IReadOnlyList<Seat> SuggestedSeats => seats;

        public IEnumerable<string> SeatNames()
        {
            return seats.Select(s => s.ToString());
        }

        public bool MatchExpectation()
        {
            return seats.Count == PartyRequested;
        }
    }
}